src/main/java/chopchop/logic/CommandDispatcher.java

```diff
@@ -0,0 +1,78 @@
+package chopchop.logic;
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.core.GuiSettings;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.AddressBookParser;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.Model;
+import seedu.address.model.ReadOnlyAddressBook;
+import seedu.address.model.person.Person;
+import seedu.address.storage.Storage;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.logging.Logger;
+
+public class CommandDispatcher implements Logic {
+    public static final String FILE_OPS_ERROR_MESSAGE = "Could not save data to file: ";
+    private final Logger logger = LogsCenter.getLogger(CommandDispatcher.class);
+
+    private final Model model;
+    private final Storage storage;
+    private final AddressBookParser addressBookParser;
+
+    /**
+     * Constructs a {@code LogicManager} with the given {@code Model} and {@code Storage}.
+     */
+    public CommandDispatcher(Model model, Storage storage) {
+        this.model = model;
+        this.storage = storage;
+        addressBookParser = new AddressBookParser();
+    }
+
+    @Override
+    public CommandResult execute(String commandText) throws CommandException, ParseException {
+        logger.info("----------------[USER COMMAND][" + commandText + "]");
+
+        CommandResult commandResult;
+        Command command = addressBookParser.parseCommand(commandText);
+        commandResult = command.execute(model);
```

Comment:
another small thing but probably it'll be easier to define the `commandResult` only when you need it (ie. on line 43 instead of 41)

Reply:
roger that